### PR TITLE
use the cache for agent config to prevent crashing when cmsweb is down

### PR DESCRIPTION
**Pull request by ticoann**
@amaltaro, Alan please review

### CODE REVIEW

File: src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py [hunk 1]
```diff
@@ -127,9 +127,12 @@ def isDrainMode(config):
 
     reqMgrAux = ReqMgrAux(config.TaskArchiver.ReqMgr2ServiceURL)
     agentConfig = reqMgrAux.getWMAgentConfig(config.Agent.hostName)
-
-    return agentConfig["UserDrainMode"] or agentConfig["AgentDrainMode"]
-
+    if "UserDrainMode" in agentConfig and "AgentDrainMode" in agentConfig:
+        AUXDB_AGENT_CONFIG_CACHE = agentConfig
```
Comment (amaltaro) [inline src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py]: Isn't it another variable/id? I think we'd need to declare it `global` if we want to overwrite its value in here, no?

**Review by amaltaro — CHANGES_REQUESTED** [review]
Thanks, Seangchan. I had it in my todo list too, but you beat me to the punch :)
Another option would be to keep the drain status cached on the component side (attr of the component), but I agree having a global cache would be better.

File: src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py (hunk 1 above)
Comment (ticoann) [reply src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py]: Yes you are right. Thanks I will update that.

**Comment by ticoann** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
This test works just fine on my VM. Triggering it one last time.

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
And that's a different test, so I assume we are Okay with this PR.

**Review by amaltaro — APPROVED** [review]

**Pull request merged**